### PR TITLE
Circ 275

### PR DESCRIPTION
**Pull request by jeremythuff**
Resolves[ CIRC-275](https://issues.folio.org/browse/CIRC-275): Allow Requests on On Order and In Process Items + Page Whitelist Tweak

This PR changes the whitelist to include the On order and In process item statuses. It also addressed allowing items with the paged status to be recalled.

### CODE REVIEW

File: src/test/java/api/requests/RequestTypeItemStatusWhiteListTests.java [hunk 1]
```diff
@@ -31,6 +31,44 @@ public void cannotCreateNoneRequestWhenItemStatusIsAnything() {
...
 
+  @Test
+  public void canCreateHoldRequestWhenItemStatusOnOrder() {
+    assertTrue(RequestTypeItemStatusWhiteList.canCreateRequestForItem(ItemStatus.ON_ORDER, RequestType.HOLD));
+  }
+
+  @Test
+  public void canCreateRecallRequestWhenItemStatusOnOrder() {
+    assertTrue(RequestTypeItemStatusWhiteList.canCreateRequestForItem(ItemStatus.ON_ORDER, RequestType.RECALL));
+
```
Comment (marcjohnson-kint) [inline src/test/java/api/requests/RequestTypeItemStatusWhiteListTests.java]: This blank line can be removed

File: src/test/java/api/requests/RequestTypeItemStatusWhiteListTests.java [hunk 2]
```diff
@@ -31,6 +31,44 @@ public void cannotCreateNoneRequestWhenItemStatusIsAnything() {
...
+  @Test
+  public void canCreateHoldRequestWhenItemStatusInProcess() {
+    assertTrue(RequestTypeItemStatusWhiteList.canCreateRequestForItem(ItemStatus.IN_PROCESS, RequestType.HOLD));
+  }
+
+  @Test
+  public void canCreateRecallRequestWhenItemStatusInProcess() {
+    assertTrue(RequestTypeItemStatusWhiteList.canCreateRequestForItem(ItemStatus.IN_PROCESS, RequestType.RECALL));
+
```
Comment (marcjohnson-kint) [inline src/test/java/api/requests/RequestTypeItemStatusWhiteListTests.java]: This blank line can be removed

File: src/main/java/org/folio/circulation/domain/ItemStatus.java [hunk 3]
```diff
@@ -11,7 +11,9 @@
   CHECKED_OUT("Checked out"),
   IN_TRANSIT("In transit"),
   MISSING("Missing"),
-  PAGED("Paged");
+  PAGED("Paged"),
+  ON_ORDER("On order"),
+  IN_PROCESS("In Process");
```
Comment (marcjohnson-kint) [inline src/main/java/org/folio/circulation/domain/ItemStatus.java]: The typical capitalisation in FOLIO is only the first word of a phrase. And the [examples in mod-orders](https://github.com/folio-org/mod-orders/blob/223580178d46c45d13479db68fc350f521b7da6a/src/test/java/org/folio/rest/impl/OrdersImplTest.java#L3665) seem to reflect this.

Was `In Process` specified this way in the story?

File: src/test/java/api/requests/RequestsAPICreationTests.java [hunk 4]
```diff
@@ -1168,24 +1168,25 @@ public void cannotCreateRecallRequestWhenItemIsMissing()
...
       .recall()
       .forItem(pagedItem)
       .withPickupServicePointId(requestPickupServicePoint.getId())
       .by(usersFixture.jessica()));
 
-    assertThat(recallResponse.getJson(), hasErrorWith(allOf(
-      hasMessage("Recall requests are not allowed for this patron and item combination"),
-      hasParameter("requestType", "Recall"))));
+      assertThat(recallResponse.getJson().getString("requestType"), is(RequestType.RECALL.getValue()));
+      assertThat(pagedItem.getResponse().getJson().getJsonObject("status").getString("name"), is(ItemStatus.PAGED.getValue()));
```
Comment (marcjohnson-kint) [inline src/test/java/api/requests/RequestsAPICreationTests.java]: I typically statically import constants / enum values, I find it improves the readability of tests

Comment (jeremythuff) [reply src/test/java/api/requests/RequestsAPICreationTests.java]: There are several references to these enums throughout. Would you suggest changing them all or only the ones that I added?

Comment (marcjohnson-kint) [reply src/test/java/api/requests/RequestsAPICreationTests.java]: Personally, I'd use the static import for all of them (IntelliJ just does this when applying that refactoring).

I likely missed them in earlier reviews or have changed my stylistic opinions :-/

**Pull request merged**